EventFilter/CTPPSRawToDigi/src/RawToDigiConverter.cc

```diff
@@ -296,6 +296,99 @@ void RawToDigiConverter::Run(const VFATFrameCollection &coll, const TotemDAQMapp
   }
 }
 
+//----------------------------------------------------------------------------------------------------
+
+void RawToDigiConverter::Run(const VFATFrameCollection &coll, const TotemDAQMapping &mapping, const TotemAnalysisMask &mask,
+      edm::DetSetVector<TotemTimingDigi> &digi, edm::DetSetVector<TotemVFATStatus> &status)
+{
+  // structure merging vfat frame data with the mapping
+  map<TotemFramePosition, Record> records;
+
+  // common processing - frame validation 
+  RunCommon(coll, mapping, records);
+  
+  // second loop over data
+  for (auto &p : records)
+  {
+    Record &record = p.second;
+    if (!record.status.isOK()) continue;
+    
+    const TotemFramePosition* framepos = &p.first;
+
+    if(((framepos->getIdxInFiber()%2)==0)&&(framepos->getIdxInFiber()<14))
+    {
+      //corresponding channel data are always in the neighbouring idx in fiber
+
+      TotemFramePosition frameposdata(framepos->getSubSystemId(),framepos->getTOTFEDId(),framepos->getOptoRxId(),framepos->getGOHId(),(framepos->getIdxInFiber()+1));
+      TotemFramePosition frameposEvtInfo(framepos->getSubSystemId(),framepos->getTOTFEDId(),framepos->getOptoRxId(),framepos->getGOHId(),0xe);
+
+      auto channelwaveformPtr = records.find(frameposdata);
+      auto eventInfoPtr = records.find(frameposEvtInfo);
+      
+      if ( channelwaveformPtr != records.end() && eventInfoPtr != records.end() ) 
+      {
+        Record &channelwaveform = records[frameposdata];
+        Record &eventInfo = records[frameposEvtInfo];  
+
+
+        // Extract all the waveform information from the raw data
+        TotemSampicFrame totemSampicFrame((uint8_t*) record.frame->getData(), (uint8_t*) channelwaveform.frame->getData(), (uint8_t*) eventInfo.frame->getData());
```

Comment:
"const qualifier was removed via a cast, this may result in thread-unsafe code", see https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22796/27190/llvm-analysis/report-04fc7f.html#EndPath